test/defer_test.go

```diff
@@ -23,3 +23,82 @@ func TestDeferOthersDisabled(t *testing.T) {
 		Arguments: []interface{}{[]interface{}{"loop"}},
 	})
 }
+
+func TestDemonstrateIneffectiveDefers(t *testing.T) {
```

Comment:
Usually this kind of tests are not part of rules tests. You could add all these cases to `testdata/defer.go`

Reply:
yea, my goal was to provide _executable_ evidence that the patterns being detected behave the way the code / warnings describe.  these aren't linted, they're run as part of the test suite.

I'm entirely fine just removing them if you'd prefer to keep the tests more focused on running the linter or something, I wrote them half to re-prove it to myself :)